source/game_sa/Animation/AnimSequenceFrames.h

```diff
@@ -31,63 +33,25 @@ union Vec16 { // 1024
  * */
 
 struct KeyFrame {
-    CQuaternion rotation;
-    float deltaTime; // relative to previous key frame
+    CQuaternion Rot;
+    float       DeltaTime; //< Relative to previous key frame
 };
 
 struct KeyFrameTrans : KeyFrame {
-    CVector translation;
+    CVector Trans;
 };
 
 struct KeyFrameCompressed {
-    int16 rot[4];    // 4096
-    int16 deltaTime; // 60
+    FixedQuat<int16, 4096.f> Rot;
+    FixedFloat<int16, 60.f>  DeltaTime;
 
-    void GetRotation(CQuaternion* quat) {
-        float scale = 1.0f / 4096.0f;
-        quat->x = float(rot[0]) * scale;
-        quat->y = float(rot[1]) * scale;
-        quat->z = float(rot[2]) * scale;
-        quat->w = float(rot[3]) * scale;
-    }
-
-    void SetRotation(const CQuaternion& quat) {
-        rot[0] = int16(quat.x * 4096.0f);
-        rot[1] = int16(quat.y * 4096.0f);
-        rot[2] = int16(quat.z * 4096.0f);
-        rot[3] = int16(quat.w * 4096.0f);
-    }
-
-    [[nodiscard]] float GetDeltaTime() const {
-        return (float)deltaTime / 60.0f;
-    }
-
-    void SetTime(float time) {
-        deltaTime = int16(time * 60.0f + 0.5f);
+    void SetTime(float compressedTime) {
+        DeltaTime.Set(compressedTime, true); // NOTE/TODO: I don't really think rounding upwards matters that much, but thats how the code was
```

Comment:
no need to be inexact without a great reason ;)
```suggestion
        DeltaTime.Set(compressedTime + (0.5f / 60.0f), true);
```

> c * 60 + 0.5
> c * 60 + (0.5 / 60) * 60
> 60 * (c + 0.5 / 60)

Reply:
`.Set` does rounding, hence that `true` there.

Reply:
then why do you have the todo?

Reply:
forgot to remove it